### PR TITLE
CVSL-386 audit refactor and first/last names into COM details

### PR DESCRIPTION
**Pull request by timharrison-moj**
PLEASE DO NOT MERGE - requires sequencing with API change.

* Removes audit events from the UI for all `change` type events on a licence.
* Create, Submit, Approve, Reject, Edit and Activate are all audited in the API now
* Retains `view` type audit events in the UI (we don't know the reason for the fetchLicence in the API).
* Conditionally audits `view` type events when the current user is NOT the responsible officer for a licence.
* Adds first / last names into the COM details request (to match the API changes which accompany this PR)
* Adjusts all tests accordingly.

### CODE REVIEW

File: server/listeners/eventHandlers/probationEvents/offenderManagerChangedEventHandler.ts [hunk 1]
```diff
@@ -5,6 +5,10 @@ import LicenceService from '../../../services/licenceService'
 export default class OffenderManagerChangedEventHandler {
   constructor(private readonly communityService: CommunityService, private readonly licenceService: LicenceService) {}
 
+  // TODO: I suspect this is an incorrect interpretation of the meaning of this event?
+  // The event occurs when the offender manager for this CRN is changed i.e. A different staff member is allocated.
+  // It is probably not raised for a change of details (e.g. email, name) for the existing offender manager?
```
[the remark this answers is not in the document]
Comment (smcveigh941) [reply server/listeners/eventHandlers/probationEvents/offenderManagerChangedEventHandler.ts:10]: The COM details (name, email etc) will also get updated when that COM logs in

File: server/routes/viewingLicences/handlers/viewLicence.ts [hunk 2]
```diff
@@ -17,13 +17,16 @@ export default class ViewAndPrintLicenceRoutes {
...
-        `Viewed licence for ${licence.forename} ${licence.surname}`,
-        `Viewed licence ID ${licence.id} type ${licence.typeCode} version ${licence.version}`,
-        licence.id,
-        new Date(),
-        user
-      )
+      if (licence?.comStaffId !== user?.deliusStaffIdentifier) {
+        // Recorded here as we do not know the reason for fetchLicence in the API
+        await this.licenceService.recordAuditEvent(
+          `Licence viewed for ${licence.forename} ${licence.surname}`,
```
Comment (smcveigh941) [inline server/routes/viewingLicences/handlers/viewLicence.ts:23]: Do you need this in `server/routes/varyingLicences/handlers/viewVariation.ts`?

File: server/listeners/eventHandlers/probationEvents/offenderManagerChangedEventHandler.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (timharrison-moj) [reply server/listeners/eventHandlers/probationEvents/offenderManagerChangedEventHandler.ts:10]: This is one of the things for discussion this afternoon. I've looked at the events for probation, and I think there are two very similar - one to inform of updated offender manager details and another to inform of a reassignment of offender manager. Just a comment for now, but discussion will clarify.

File: server/routes/viewingLicences/handlers/viewLicence.ts (hunk 2 above)
Comment (timharrison-moj) [reply server/routes/viewingLicences/handlers/viewLicence.ts:23]: There will be an ongoing piece to add audits where required, as not all covered yet. Follow-on piece?

Comment (smcveigh941) [reply server/routes/viewingLicences/handlers/viewLicence.ts:23]: Team members can view and create the variation on behalf of a COM

**Review by smcveigh941 — APPROVED** [review]
LGTM

**Pull request merged**